src/base/bittorrent/session.cpp

```diff
@@ -2063,6 +2063,20 @@ bool Session::addTorrent_impl(CreateTorrentParams params, const MagnetUri &magne
             patchedFastresumeData.replace("6:pausedi0e", "6:pausedi1e");
             patchedFastresumeData.replace("12:auto_managedi0e", "12:auto_managedi1e");
 
+            // converting relative save_path to absolute
+            int start = patchedFastresumeData.indexOf("9:save_path") + 11;
+            if (start > -1) {
+                int end = patchedFastresumeData.indexOf(":", start);
+                int len = patchedFastresumeData.mid(start, end - start).toInt();
+                if (len > 0) {
+                    QByteArray oldPath = patchedFastresumeData.mid(end + 1, len);
+                    QByteArray newPath = Profile::instance().fromPortablePath(Utils::Fs::toUniformPath(fromLTString(oldPath))).toUtf8();
+                    if (oldPath != newPath)
+                        patchedFastresumeData.replace(QByteArray().append("9:save_path").append(QString::number(len).toUtf8()).append(":").append(oldPath),
```

Comment:
This expression looks horrible. Please create additional (const) variables to hold encoded paths.

Reply:
Created additional const variables for replacing pieces.